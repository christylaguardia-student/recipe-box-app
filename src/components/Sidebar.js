import React, { Component } from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import { request } from '../services/recipe-box.api';

export default class SideBar extends Component {

  constructor(props) {
    super(props);

    this.state = {
      sidebarRecipeList: []
    }
  }

  componentDidMount() {
    return request.getAll()
      .then(recipes => {
        this.setState({ sidebarRecipeList: recipes });
      })
      .catch(error => console.log('uh-oh! there was an error getting all the recipes.'));
  }

  render() {

    return (
      <Router>
        <div>
          <h2>Recipes</h2>
          <ul>
            <li><Link to={'/recipes'}>view a recipe</Link></li>
            {this.state.sidebarRecipeList.map(item => {
              return <li key={item._id}><Link to={`/recipes/${item._id}`}>{item.title}</Link></li>
            })}
          </ul>
        </div>
      </Router>
    );
  }
}
