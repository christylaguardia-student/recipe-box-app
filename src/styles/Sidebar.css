
#side-nav {
  min-height: 90vh;
  background-color: #f1f1f1;
  padding: .5rem;
}

#side-nav ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: auto;
}

#side-nav li a {
  display: block;
  color: #000;
  padding: 8px 16px;
  text-decoration: none;
}

/* TODO: active class */

#side-nav li a.active {
  background-color: #fff;
  color: white;
}

#side-nav li a:hover:not(.active) {
  background-color: #555;
  color: #fff;
}
