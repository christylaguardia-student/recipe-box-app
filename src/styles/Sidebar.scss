@import "/_colors";

#side-nav {
  min-height: calc(100vh - 50px);
  background-color: $white;
  padding: .5rem;
  border-right: 1px solid $dark;
}

#side-nav ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: auto;
}

#side-nav li a {
  display: block;
  color: $black;
  padding: 8px 16px;
  text-decoration: none;
}

#side-nav li a.active {
  background-color: #8F8F8F;
}

#side-nav li a:hover:not(.active) {
  background-color: $highlight;
  color: $white;
}
