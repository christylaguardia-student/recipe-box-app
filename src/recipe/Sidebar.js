import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import '../styles/Sidebar.css';
import { getRecipe } from './actions'; 

class SideBar extends Component {

  render() {
    const { all } = this.props.recipes;
    const recipes = all;

    return (
      <div id="side-nav">
        <ul>
          {recipes.map(item => {
            return (
              <li key={item._id}>
                <NavLink to={`/recipes/${item._id}`}>{item.title}</NavLink>
              </li>
            );
          })}
        </ul>
      </div>
    );
  }
}

export default connect(state => {
  return { recipes: state.recipes };
}, { getRecipe })(SideBar);
