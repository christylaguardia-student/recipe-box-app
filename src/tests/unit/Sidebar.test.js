import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Sidebar from '../../components/Sidebar';

describe('<Sidebar />', () => {

  const testRecipes = [
    {
      "_id": 1,
      "title": "Chocolate Chip Cookies",
      "instructions": "Mix stuff and put on cookie sheet. Then put in oven. Do not burn.",
      "ingredients": [ 
        { "_id": 1, "name": "flour", "amount": 2, "unit": "cup" },
        { "_id": 2, "name": "sugar", "amount": 1, "unit": "cup" },
        { "_id": 3, "name": "chocolate chips", "amount": 2.5, "unit": "cup" }
      ]
    },{
      "_id": 2,
      "title": "White Chocolate Chip Cookies",
      "instructions": "Mix stuff and put on cookie sheet. Then put in oven. Do not burn.",
      "ingredients": [ 
        { "_id": 1, "name": "flour", "amount": 2, "unit": "cup" },
        { "_id": 2, "name": "sugar", "amount": 1, "unit": "cup" },
        { "_id": 3, "name": "white chocolate chips", "amount": 2.5, "unit": "cup" }
      ]
    }
  ]

  it('renders the Sidebar component', () => {
    const wrapper = toJSON(shallow(<Sidebar sidebarRecipeList={testRecipes} />));
    expect(wrapper).toMatchSnapshot();
  });

})