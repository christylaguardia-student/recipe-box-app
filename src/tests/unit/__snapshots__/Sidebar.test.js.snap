// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders the Sidebar component 1`] = `
<div
  id="side-nav"
>
  <ul />
</div>
`;
